### PR TITLE
Add Github On-Perm Import Script

### PR DESCRIPTION
**Pull request by gtchuinkam**

### CODE REVIEW

File: snippets/scripts/compass-github-importer/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+## Bootstrap your Compass Catalog with from a Self-Hosted GitHub Instance
+
+This script
+1. Iterates over all your GitHub organizations and all the repos within them
+2. Creates Compass Components for each of them
```
Comment (rvenkatesh-atl) [inline snippets/scripts/compass-github-importer/README.md:5]: Apart from creating a component (with associated repo link), do we include any additional meta-data i.e repo description, labels etc?

Comment (rvenkatesh-atl) [reply snippets/scripts/compass-github-importer/README.md:5]: nvm I see below it includes "description", probably worth expanding on what import means in that sense i.e we create component and update description etc.

File: snippets/scripts/compass-github-importer/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+If your connection is interrupted, or you want to re-run after the initial import you can. It checks for duplicates and skips them:
+
+```
+Already added https://github.com/VishnuprakashJ/home ... skipping
+```
+
+## Next steps
+
+Once you're finished set up the on-prem webhook to link deployments, builds, etc to Compass. You only have to do this once (not once per Component) and Compass will associate the repo events with the Component they interact with.
```
Comment (shrestabangaru) [inline snippets/scripts/compass-github-importer/README.md:80]: Maybe specify: 
"once per organization (not per Component)"

File: snippets/scripts/compass-github-importer/index.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+          }
+        }
+      }
+      `,
+    });
+    const maybeResults = response?.data?.compass?.searchComponents?.nodes;
+
+    if (!Array.isArray(maybeResults)) {
+        console.error(`error fetching component: `, JSON.stringify(response));
+        throw new Error('Error fetching component');
```
Comment (shrestabangaru) [inline snippets/scripts/compass-github-importer/index.js:50]: Do we want to throw with some additional details here like the componentId, organizationId (non UGC data that is available)?

File: snippets/scripts/compass-github-importer/README.md (hunk 2 above)
Comment (gtchuinkam) [reply snippets/scripts/compass-github-importer/README.md:80]: I do believe the webhooks are per app, so they would only have to do it once for this GitHub import, so I think the wording here is okay.

File: snippets/scripts/compass-github-importer/index.js (hunk 3 above)
Comment (gtchuinkam) [reply snippets/scripts/compass-github-importer/index.js:50]: Good, call will do!

File: snippets/scripts/compass-github-importer/index.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        }
+    }
+}
+
+async function listAllOrgs() {
+    let instanceOrganizations = []
+    let page = 1;
+    const perPage = 100; // Maximum items per page as allowed by GitHub Enterprise Server API
+
+    while(true) {
```
Comment (shrestabangaru) [inline snippets/scripts/compass-github-importer/index.js:96]: Is there any way we can replace this with a for loop based on `page`, or avoid the "while true"?

**Review by shrestabangaru — APPROVED** [review]

File: snippets/scripts/compass-github-importer/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+New component for https://github.com/hyde.me/test_gb_less3_2 ... would be added hyde.me/test_gb_less3_2 (dry-run)
+New component for https://github.com/sachintomar009/reposetup ... would be added sachintomar009/reposetup (dry-run)
+New component for https://github.com/brianwilliamaldous/api-la ... would be added brianwilliamaldous/api-la (dry-run)
+New component for https://github.com/cloud-group3072118/timely ... would be added cloud-group3072118/timely (dry-run)
+....
+
+```
+
+
+By default, all repos will be added. If you see some repos you don't want to import during the dry-run? Modify `index.js` to add some filters.
```
Comment (eserranovalle) [inline snippets/scripts/compass-github-importer/README.md:51]: Nit: the phrasing there can be simplified a bit (we currently have "If" and a question mark). A possible alternative:
> If you notice any repositories that you don't want to import during the dry run, you can modify index.js to add filters accordingly.

**Review by shrestabangaru — APPROVED** [review]

**Pull request merged**